infrastructure/src/main/java/org/corfudb/infrastructure/SequencerServer.java

```diff
@@ -128,7 +129,9 @@
     private long maxConflictWildcard = Address.NOT_FOUND;
 
     private final long defaultCacheSize = Long.MAX_VALUE;
-    private final Cache<Integer, Long> conflictToGlobalTailCache;
+    private final long defaultPerStreamCacheSize = Long.MAX_VALUE;
+    private long perStreamCacheSize = defaultPerStreamCacheSize;
+    private final Cache<UUID, Cache<Integer, Long>> conflictToGlobalTailCache;
```

Comment:
I worry about these cache in cache kind of things. The cache is not passive entiry it does it's own evictions, metrics etc. Is it possible to keep a simpler passive data structure like a HashMap of Caches or just a cache<Stream, Pair<Stream, conflictHash>> or something to that effect. This 2 level nesting of the cache data structure  makes me uncomfortable.